utils/utils.h

```diff
@@ -124,9 +124,17 @@
 
 # define  MSG(fd, fmt, ...)                                          \
           do {                                                       \
+            int len;                                                 \
+            char *buf;                                               \
             if (fd <= 0)       /* including STDIN_FILENO 0 */        \
               fd = stderr;                                           \
-            fprintf(fd, fmt, __VA_ARGS__);                           \
+            len = GB_ASPRINTF(&buf, fmt, ##__VA_ARGS__);             \
+            if (len != -1) {                                         \
+              if (buf[len - 1] != '\n')                              \
+                buf[len - 1] = '\n';                                 \
+              fprintf(fd, "%s", buf);                                \
+              free(buf);                                             \
+            }                                                        \
```

Comment:
@pkalever 

BTW, should we just assume that it doesn't have the '\n' when the GB_ASPRINTF fails and print the logs by using:

fprintf(fd, fmt "\n", ##__VA_ARGS__); 

Or just do nothing as above ?

Thanks.

Reply:
Yeah, obviously something is better than nothing :-)
Lets print what's there in the 'fmt + __VA_ARGS__', when GB_ASPRINTF fail.

Thanks!

Reply:
Updated it.
Thanks.

Reply:
Looks good to me.
@lxbsz I also feel like changing line 138 to
fprintf(fd, fmt "\n", ##__VA_ARGS__);

won't hurt. Coz not having a newline looks awkward than having two '\n'.

Reviewed-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>